1-select-tasks/22.txt

```diff
@@ -1,3 +1,15 @@
-Create a SQL query to display the name of each customer along with their most expensive purchased product:
+--22Create a SQL query to display the name of each customer along with their most expensive purchased product:
 
-| CompanyName | ProductName | PricePerItem |
\ No newline at end of file
+--| CompanyName | ProductName | PricePerItem |
+
+SELECT [c].[CompanyName], [p].[ProductName], MAX([p].[UnitPrice])  AS [PricePerItem]
+FROM [dbo].[Customers][c]
+JOIN [dbo].[Orders][o] ON [o].[CustomerID] = [c].[CustomerID]
+JOIN [dbo].[Order Details][od] ON [od].[OrderID] = [o].[OrderID]
+JOIN [dbo].[Products][p] ON [p].[ProductID] = [od].[ProductID]
+GROUP BY [c].[CustomerID], [c].[CompanyName],[p].[ProductName], [p].[UnitPrice]
```

Comment:
what do you need the group by clause for?

Reply:
before HAVING use, and used aggregate functions.